tests/test_ecs/test_ecs_boto3.py

```diff
@@ -1705,6 +1705,109 @@ def test_run_task():
     response["tasks"][0]["tags"][0].get("value").should.equal("tagValue0")
 
 
+@mock_ec2
+@mock_ecs
+def test_run_task_awsvpc_network():
+
+    # Setup
+    client = boto3.client("ecs", region_name="us-east-1")
+    ec2_client = boto3.client("ec2", region_name="us-east-1")
+    ec2 = boto3.resource("ec2", region_name="us-east-1")
+
+    # ECS setup
+    setup_resources = setup_ecs(client, ec2)
+
+    # Execute
+    response = client.run_task(
+        cluster="test_ecs_cluster",
+        overrides={},
+        taskDefinition="test_ecs_task",
+        startedBy="moto",
+        launchType="FARGATE",
+        networkConfiguration={
+            "awsvpcConfiguration": {
+                "subnets": [setup_resources[0].id],
+                "securityGroups": [setup_resources[1].id],
+            }
+        },
+    )
+
+    # Verify
+    len(response["tasks"]).should.equal(1)
+    response["tasks"][0]["lastStatus"].should.equal("RUNNING")
+    response["tasks"][0]["desiredStatus"].should.equal("RUNNING")
+    response["tasks"][0]["startedBy"].should.equal("moto")
+    response["tasks"][0]["stoppedReason"].should.equal("")
+
+    eni = ec2_client.describe_network_interfaces(
+        Filters=[{"Name": "description", "Values": ["moto ECS"]}]
+    )["NetworkInterfaces"][0]
+    try:
+        # should be UUID
+        UUID(response["tasks"][0]["attachments"][0]["id"])
+    except ValueError:
+        assert False
+    response["tasks"][0]["attachments"][0]["status"].should.equal("ATTACHED")
+    response["tasks"][0]["attachments"][0]["type"].should.equal(
+        "ElasticNetworkInterface"
+    )
+    details = response["tasks"][0]["attachments"][0]["details"]
+
+    subnet_found = False
```

Comment:
This entire section may be more readable if we use:
```
assert {"name": "subnetId", "value": setup_resources[0].id} in details
assert {"name": "networkInterfaceId", ..} in details
# etc
```
As far as I can tell, that would test the same thing

Reply:
You're right, much cleaner and less lines 💪